bothan-lib/src/worker/rest.rs

```diff
@@ -41,30 +43,53 @@ pub async fn start_polling<S: Store, E: Display, P: AssetInfoProvider<Error = E>
 
         select! {
             _ = cancellation_token.cancelled() => break,
-            r = timeout(interval.period(), provider.get_asset_info(&ids)) => handle_polling_result(r, &store).await,
+            r = handle_get_asset_info(&provider, &ids, update_interval, meter_name) => handle_polling_result(r, &store, meter_name).await,
         }
     }
 }
 
+async fn handle_get_asset_info<P, E>(
+    provider: &P,
+    ids: &Vec<String>,
+    timeout_interval: Duration,
+    meter_name: &'static str,
+) -> Result<Result<Vec<AssetInfo>, E>, Elapsed>
+where
+    E: Display,
+    P: AssetInfoProvider<Error = E>,
+{
+    let start_time = chrono::Utc::now().timestamp_millis();
+    let result = timeout(timeout_interval, provider.get_asset_info(ids)).await;
+    match result {
+        Ok(_) => Metrics::record_rest_get_asset_info_latency(meter_name, start_time, "success"),
+        Err(_) => Metrics::record_rest_get_asset_info_latency(meter_name, start_time, "failed"),
+    }
```

Comment:
why use match here? the only difference is "success" or "failed". id be much more succinct to do something like 

`let status  = result.is_err()` or if we're still using status use a if else statement

then call fn(..., status)